src/app/drive/services/thumbnail.service.ts

```diff
@@ -0,0 +1,168 @@
+import { StorageTypes } from '@internxt/sdk/dist/drive';
+import analyticsService from '../../analytics/services/analytics.service';
+import { AppView, DevicePlatform } from '../../core/types';
+import localStorageService from '../../core/services/local-storage.service';
+import navigationService from '../../core/services/navigation.service';
+import { getEnvironmentConfig } from './network.service';
+import { SdkFactory } from '../../core/factory/sdk';
+import { uploadFile as uploadToBucket } from 'app/network/upload';
+import notificationsService, { ToastType } from '../../notifications/services/notifications.service';
+import { Thumbnail } from '@internxt/sdk/dist/drive/storage/types';
+import { thumbnailableExtension, thumbnailableImageExtension, thumbnailablePdfExtension } from 'app/drive/types/file-types';
+import { FileToUpload } from './file.service/uploadFile';
+import Resizer from 'react-image-file-resizer';
+import { DriveItemData, ThumbnailConfig } from '../types';
+import fetchFileBlob from './download.service/fetchFileBlob';
+import { Downloadable } from 'app/network/download';
+import { storageActions } from 'app/store/slices/storage';
+import { AppDispatch } from 'app/store';
+
+
+export interface ThumbnailToUpload {
+  fileId: number;
+  size: number;
+  type: string;
+  content: File;
+}
+
+const getImageThumbnail = async (file: File): Promise<File | null> => {
+  return await new Promise((resolve) => {
```

Comment:
Here you can remove async and await to achieve the same functioanlity

Reply:
Also does Resizer.imageFileResizer ever fail? if passed a broken image perhaps is good to catch the error in this Promise `reject` function